index.bs

```diff
@@ -535,101 +539,233 @@ dictionary WriteParams {
+Upon creation, an underlying sink will have been created and the stream will be usable.
+All operations executed on the stream are queuable and producers will be able to respond to backpressure.
 
-     If |data| is of type {{WriteParams}}, validates and executes the content of |data|, which might result in changes into the file associated with |stream|.
+The underlying sink's write method, and therefore {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}}
+method, will accept byte-like data or {{WriteParams}} as input.
 
-     {{WriteParams}} requires the attribute {{WriteParams/type|type}} which determines what the operations to
-     execute are.
+The {{FileSystemWritableFileStream}} has a file position cursor initialized at byte offset 0 from the top of the file.
+When using {{FileSystemWritableFileStream/write()|write()}} or by using WritableStream capabilities through the {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, this position will be advanced based on the number of bytes written through the stream object.
 
-     If {{WriteParams/type|type}} is {{truncate}}, {{WriteParams/size|size}} is a <span class=allow-2119>required</span> attribute.
+Similarly, when piping a {{ReadableStream}} into a {{FileSystemWritableFileStream}} object, this position is updated with the number of bytes that passed through the stream.
 
-     If {{WriteParams/type|type}} is {{seek}}, {{WriteParams/position|position}} is a <span class=allow-2119>required</span> attribute.
+{{WritableStream/getWriter()|getWriter()}} returns an instance of {{WritableStreamDefaultWriter}}.
+</div>
 
-     If {{WriteParams/type|type}} is {{write}}, {{WriteParams/position|position}} is an <span class=allow-2119>optional</span> attribute, {{WriteParams/data|data}} is a <span class=allow-2119>required</span> attribute.
+<div algorithm>
+To <dfn>create a new FileSystemWritableFileStream</dfn> given a [=file entry=] `file`, perform the following steps:
+
+1. Let |stream| be a new {{FileSystemWritableFileStream}}.
+1. Perform [$InitializeWritableStream$](|stream|)
+1. Set |stream|.[=FileSystemWritableFileStream/[[file]]=] to [=file entry=].
+1. Let |controller| be a new {{WritableStreamDefaultController}}.
+1. Let |startAlgorithm| be an algorithm that returns `undefined`.
+1. Let |writeAlgorithm| be an algorithm which takes a |chunk| argument
+   and runs the [=write a chunk=] algorithm with |stream| and |chunk|.
+1. Let |closeAlgorithm| be the following steps:
+   1. Issue: TODO: Check permissions.
+   1. Set |stream|.[=[[file]]=]'s [=file entry/binary data=] to |stream|.[=[[buffer]]=].
+      If that throws an exception, return [=a promise rejected with=] that exception.
+
+      Note: it is expected that this atomically updates the contents of the file on disk
+      being written to.
+1. Let |abortAlgorithm| be an algorithm that returns `undefined`.
```

Comment:
The return value has to be strictly a promise, so it needs to be "an algorithm that returns a promise resolved with undefined".

Reply:
It's pretty arbitrary. Mostly it was just easier to spec that way. One justification is that other specifications rarely need to do anything in the start method, so it's easier for them to just return undefined.